lib/puppet/provider/cisco_ace/nxapi.rb

```diff
@@ -0,0 +1,222 @@
+        current_state[prop] = nil
+      else
+        current_state[prop] = val ? :true : :false
+      end
+    end
+    debug current_state
+    new(current_state)
+  end # self.properties_get
+
+  def self.instances
+    ace_hash = []
+    Cisco::Ace.aces.each do |afi, acl_aces|
+      acl_aces.each do |acl_name, aces|
+        aces.each do |seqno, ace_instance|
+          ace_hash << properties_get(afi, acl_name, seqno, ace_instance)
+        end
+      end
+    end
+    ace_hash
+  end
+
+  def self.prefetch(resources)
+    ace_instances = instances
+    resources.keys.each do |name|
+      provider = ace_instances.find do |ace|
+        resources[name][:seqno] == ace.seqno.to_i &&
+        resources[name][:acl_name] == ace.acl_name.to_s &&
+        resources[name][:afi] == ace.afi.to_s
+      end
+      resources[name].provider = provider unless provider.nil?
+    end
+  end # self.prefetch
+
+  def exists?
+    @property_hash[:ensure] == :present
+  end
+
+  def create
+    @property_flush[:ensure] = :present
+  end
+
+  def destroy
+    @property_flush[:ensure] = :absent
+  end
+
+  def properties_set(new_ace=false)
+    ACL_ALL_PROPS.each do |prop|
+      next unless @resource[prop]
+      if new_ace
+        # Set @property_flush for the current object
+        send("#{prop}=", @resource[prop])
+      end
+      next if @property_flush[prop].nil?
+      # Call the AutoGen setters for the @ace
+      # node_utils object.
+      @ace.send("#{prop}=", @property_flush[prop]) if
+        @ace.respond_to?("#{prop}=")
+    end
+    # Set methods that are not autogenerated follow.
+    ace_set
+  end
+
+  # Custom setters.
+  # The following properties are setters and cannot be handled
+  # by PuppetX::Cisco::AutoGen.mk_puppet_methods.
+  def ace_set
+    return unless @property_flush[:action] ||
+                  @property_flush[:proto] ||
+                  @property_flush[:src_addr] ||
+                  @property_flush[:src_port] ||
+                  @property_flush[:dst_addr] ||
+                  @property_flush[:dst_port]
+    @ace.ace_set(@property_flush)
+  end
+
+  def action_set
+    return unless @property_flush[:action]
+    @ace.action = @property_flush[:action]
+  end
+
+  def proto_set
+    return unless @property_flush[:proto]
+    @ace.proto = @property_flush[:proto]
+  end
+
+  def src_addr_set
+    return unless @property_flush[:src_addr]
+    @ace.src_addr = @property_flush[:src_addr]
+  end
+
+  def src_port_set
+    return unless @property_flush[:src_port]
+    @ace.src_port = @property_flush[:src_port]
+  end
+
+  def dst_addr_set
+    return unless @property_flush[:dst_addr]
+    @ace.dst_addr = @property_flush[:dst_addr]
+  end
+
+  def dst_port_set
+    return unless @property_flush[:dst_port]
+    @ace.dst_port = @property_flush[:dst_port]
+  end
+
+  def option_format_set
+    return unless @property_flush[:option_format]
+    @ace.option_format = @property_flush[:option_format]
+  end
+
+  def flush
```

Comment:
As we discussed yesterday, you need to handle the absent / destroy case.

Example code here: 
https://github.com/cisco/cisco-network-puppet-module/blob/develop/lib/puppet/provider/cisco_bgp_neighbor_af/nxapi.rb#L314